ert_shared/ensemble_evaluator/prefect_ensemble/storage_driver.py

```diff
@@ -12,25 +12,29 @@ def storage_driver_factory(config, run_path):
 
 class _SharedDiskStorageDriver:
     def __init__(self, storage_path, run_path):
-        self._storage_path = storage_path
-        self._run_path = f"{run_path}"
+        self._storage_path = Path(storage_path)
+        self._run_path = Path(f"{run_path}")
 
     def get_storage_path(self, iens):
         if iens is None:
-            return f"{self._storage_path}/global"
-        return f"{self._storage_path}/{iens}"
+            return self._storage_path / "global"
+        return self._storage_path / str(iens)
 
     def store(self, local_name, iens=None):
         storage_path = self.get_storage_path(iens)
-        os.makedirs(storage_path, exist_ok=True)
-        storage_uri = os.path.join(storage_path, local_name)
-        shutil.copyfile(os.path.join(self._run_path, local_name), storage_uri)
+        storage_path.mkdir(parents=True, exist_ok=True)
+        storage_uri = storage_path / local_name
+        shutil.copyfile(self._run_path / local_name, storage_uri)
         return storage_uri
 
     def retrieve(self, storage_uri, target=None):
-        if storage_uri.startswith(self._storage_path):
-            target = os.path.basename(storage_uri) if target is None else target
-            shutil.copyfile(storage_uri, os.path.join(self._run_path, target))
-            return target
-        else:
-            raise ValueError(f"Storage driver can't handle file: {storage_uri}")
+        storage_uri = Path(storage_uri)
+        try:
+            storage_uri.relative_to(self._storage_path)
```

Comment:
This was a bit peculiar... Care to explain? :)

Reply:
Having another look, I notice that the error handling is a bit peculiar in this method. This is a direct translation of the original `os.path` code, which  checked if the `storage_uri` string starts with `self._storage_path` string, instead of just checking if the file exists... I will change that error checking code to be more sensible...